# Problem Statement:-
- Obesity is a worldwide problem which causes a lot of serious medical problems. Obesity will be increased, about 45% out of the whole population by 2035. The proportion of morbid obese and the actual costs on healthcare will be increased. Implementation of a system that could estimate the percentage of obese population for particular time duration given the age range, income range, location, high confidence level and low confidence level of obesity, education, gender, the class level, etc. of the population can help in fight against obesity.

# Importing libraries

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

In [2]:
data = pd.read_csv('Nutrition__Physical_Activity__and_Obesity_-_Behavioral_Risk_Factor_Surveillance_System.csv')
data.head()

YearStart  YearEnd LocationAbbr LocationDesc  \
0       2011     2011           AL      Alabama   
1       2011     2011           AL      Alabama   
2       2011     2011           AL      Alabama   
3       2011     2011           AL      Alabama   
4       2011     2011           AL      Alabama   

                                   Datasource                    Class  \
0  Behavioral Risk Factor Surveillance System  Obesity / Weight Status   
1  Behavioral Risk Factor Surveillance System  Obesity / Weight Status   
2  Behavioral Risk Factor Surveillance System  Obesity / Weight Status   
3  Behavioral Risk Factor Surveillance System  Obesity / Weight Status   
4  Behavioral Risk Factor Surveillance System  Obesity / Weight Status   

                     Topic                                           Question  \
0  Obesity / Weight Status  Percent of adults aged 18 years and older who ...   
1  Obesity / Weight Status  Percent of adults aged 18 years and older who ...   
2  Obesity / Weight Status  Percent of adults aged 18 years and older who ...   
3  Obesity / Weight Status  Percent of adults aged 18 years and older who ...   
4  Obesity / Weight Status  Percent of adults aged 18 years and older who ...   

   Data_Value_Unit Data_Value_Type  ...  \
0              NaN           Value  ...   
1              NaN           Value  ...   
2              NaN           Value  ...   
3              NaN           Value  ...   
4              NaN           Value  ...   

                               GeoLocation  ClassID TopicID QuestionID  \
0  (32.84057112200048, -86.63186076199969)      OWS    OWS1       Q036   
1  (32.84057112200048, -86.63186076199969)      OWS    OWS1       Q036   
2  (32.84057112200048, -86.63186076199969)      OWS    OWS1       Q036   
3  (32.84057112200048, -86.63186076199969)      OWS    OWS1       Q036   
4  (32.84057112200048, -86.63186076199969)      OWS    OWS1       Q036   

   DataValueTypeID  LocationID  StratificationCategory1  \
0            VALUE           1                    Total   
1            VALUE           1                   Gender   
2            VALUE           1                   Gender   
3            VALUE           1                Education   
4            VALUE           1                Education   

         Stratification1 StratificationCategoryId1 StratificationID1  
0                  Total                       OVR           OVERALL  
1                   Male                       GEN              MALE  
2                 Female                       GEN            FEMALE  
3  Less than high school                       EDU             EDUHS  
4   High school graduate                       EDU         EDUHSGRAD  

[5 rows x 33 columns]

In [3]:
data.shape

(53392, 33)

# Observation:- 
- There are 53392 rows and 33 columns.
- Each row have individual person information.

In [4]:
data.describe()

YearStart       YearEnd  Data_Value_Unit    Data_Value  \
count  53392.000000  53392.000000              0.0  48346.000000   
mean    2013.281465   2013.281465              NaN     31.156681   
std        1.693300      1.693300              NaN     10.247033   
min     2011.000000   2011.000000              NaN      0.900000   
25%     2012.000000   2012.000000              NaN     24.100000   
50%     2013.000000   2013.000000              NaN     30.700000   
75%     2015.000000   2015.000000              NaN     37.000000   
max     2016.000000   2016.000000              NaN     77.600000   

       Data_Value_Alt  Low_Confidence_Limit  High_Confidence_Limit   \
count    48346.000000          48346.000000            48346.000000   
mean        31.156681             26.892227               35.989997   
std         10.247033             10.038584               11.205813   
min          0.900000              0.300000                3.000000   
25%         24.100000             20.000000               28.200000   
50%         30.700000             26.450000               35.600000   
75%         37.000000             32.900000               42.200000   
max         77.600000             69.500000               87.700000   

        Sample_Size    LocationID  
count   48346.00000  53392.000000  
mean     3889.19286     30.282215  
std     19829.42129     16.821318  
min        50.00000      1.000000  
25%       566.00000     17.000000  
50%      1209.00000     30.000000  
75%      2519.00000     44.000000  
max    476876.00000     78.000000

In [5]:
# Data summary
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 53392 entries, 0 to 53391
Data columns (total 33 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   YearStart                   53392 non-null  int64  
 1   YearEnd                     53392 non-null  int64  
 2   LocationAbbr                53392 non-null  object 
 3   LocationDesc                53392 non-null  object 
 4   Datasource                  53392 non-null  object 
 5   Class                       53392 non-null  object 
 6   Topic                       53392 non-null  object 
 7   Question                    53392 non-null  object 
 8   Data_Value_Unit             0 non-null      float64
 9   Data_Value_Type             53392 non-null  object 
 10  Data_Value                  48346 non-null  float64
 11  Data_Value_Alt              48346 non-null  float64
 12  Data_Value_Footnote_Symbol  5046 non-null   object 
 13  Data_Value_Footnote         504

In [6]:
data = data.drop(['Low_Confidence_Limit','High_Confidence_Limit ', 'Total', 'Data_Value_Alt', 'YearEnd','Topic','Class','Datasource','Data_Value_Unit','QuestionID','ClassID','TopicID','DataValueTypeID','Data_Value_Type','Data_Value_Footnote_Symbol','Data_Value_Footnote','StratificationCategoryId1','StratificationID1'],1);
data = data.drop(['GeoLocation', 'Question', 'StratificationCategory1', 'Stratification1', 'Gender'],1)

In [7]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 53392 entries, 0 to 53391
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   YearStart       53392 non-null  int64  
 1   LocationAbbr    53392 non-null  object 
 2   LocationDesc    53392 non-null  object 
 3   Data_Value      48346 non-null  float64
 4   Sample_Size     48346 non-null  float64
 5   Age(years)      11438 non-null  object 
 6   Education       7628 non-null   object 
 7   Income          13349 non-null  object 
 8   Race/Ethnicity  15256 non-null  object 
 9   LocationID      53392 non-null  int64  
dtypes: float64(2), int64(2), object(6)
memory usage: 4.1+ MB


In [8]:
# Checking null values in dataset
data.isnull().sum()

YearStart             0
LocationAbbr          0
LocationDesc          0
Data_Value         5046
Sample_Size        5046
Age(years)        41954
Education         45764
Income            40043
Race/Ethnicity    38136
LocationID            0
dtype: int64

In [9]:
# Checking percentage of null value in dataset
data.isnull().sum()/len(data) * 100

YearStart          0.000000
LocationAbbr       0.000000
LocationDesc       0.000000
Data_Value         9.450854
Sample_Size        9.450854
Age(years)        78.577315
Education         85.713215
Income            74.998127
Race/Ethnicity    71.426431
LocationID         0.000000
dtype: float64

In [10]:
# Subsetting the data to be able to analyze according to specific column
data = data.dropna(subset=['Education'])
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 7628 entries, 3 to 53370
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   YearStart       7628 non-null   int64  
 1   LocationAbbr    7628 non-null   object 
 2   LocationDesc    7628 non-null   object 
 3   Data_Value      7628 non-null   float64
 4   Sample_Size     7628 non-null   float64
 5   Age(years)      0 non-null      object 
 6   Education       7628 non-null   object 
 7   Income          0 non-null      object 
 8   Race/Ethnicity  0 non-null      object 
 9   LocationID      7628 non-null   int64  
dtypes: float64(2), int64(2), object(6)
memory usage: 655.5+ KB


In [12]:
data.dtypes

YearStart           int64
LocationAbbr       object
LocationDesc       object
Data_Value        float64
Sample_Size       float64
Age(years)         object
Education          object
Income             object
Race/Ethnicity     object
LocationID          int64
dtype: object

In [14]:
# converting columns name to lower case
data.columns  = data.columns.str.lower()

### Checking for duplicate and missing value

In [17]:
data.duplicated().sum()

8

In [18]:
# dropping duplicate data
data = data.drop_duplicates()

In [19]:
data.isna().sum() 

yearstart            0
locationabbr         0
locationdesc         0
data_value           0
sample_size          0
age(years)        7620
education            0
income            7620
race/ethnicity    7620
locationid           0
dtype: int64

In [20]:
# Checking for percentage value of null values
data.isnull().sum()/len(data) * 100

yearstart           0.0
locationabbr        0.0
locationdesc        0.0
data_value          0.0
sample_size         0.0
age(years)        100.0
education           0.0
income            100.0
race/ethnicity    100.0
locationid          0.0
dtype: float64

In [ ]:
# Dropping columns with high volumn null %